London | Maryn Romanova | Module-Data-Flows |  Book Library Project

MarynaRomashca opened the pull request:
<!--

You must title your PR like this:

COHORT_NAME | FIRST_NAME LAST_NAME | REPO_NAME | WEEK

For example,

NW4 | Carol Owen | HTML-CSS-Module | Week1

Complete the task list below this message.
If your PR is rejected, check the task list.

-->

## Learners, PR Template

Self checklist

- [x] I have committed my files one by one, on purpose, and for a reason
- [x] I have titled my PR with COHORT_NAME | FIRST_NAME LAST_NAME | REPO_NAME | WEEK 
- [x] I have tested my changes
- [x] My changes follow the [style guide](https://curriculum.codeyourfuture.io/guides/contributing/)
- [x] My changes meet the [requirements](./README.md) of this task

## Changelist

Briefly explain your PR.

## Questions

Ask any questions you have for your reviewer.

jenny-alexander reviewed (changes requested):
1. Can you review all of the places that you declared a variable with 'let'? In general, you will mostly always use const (and this should be your default). If you need to reassign a value to the variable, then you would use let. Here is a video you can watch: https://youtu.be/RE6qf3As-XU?si=_YaiEPeBC5Cx_j56

2. The 4 variables on lines 24-27 are currently declared as global variables. However, they are only used within the submit function.
Can modify your code so that they are no longer global variables and are declared as local variables?
Making this change means the variables won't be created until they are needed within the submit function. This makes the code cleaner and more maintainable!
https://www.w3schools.com/js/js_scope.asp

3. Currently your application does not have a title in the browser table. Can you update the application title to something more meaningful?
<img width="225" alt="Screenshot 2025-04-20 at 4 38 38 PM" src="https://github.com/user-attachments/assets/08999cca-35f6-4197-8d66-25623f63bcdf" />

cjyuan commented:
@jenny-alexander You have given a review yesterday. So I assumed you meant to add "Reviewed" label and changed the label.

@MarynaRomashca Please respond the the reviewer's feedback and change your code accordingly. Your Step 4 will only be approved when you have completed this PR.

MarynaRomashca commented:
@jenny-alexander and @cjyuan I have change my code accordingly. @jenny-alexander please approve my  PR, or let me know if I need to do extra work.

jenny-alexander commented:
> @jenny-alexander and @cjyuan I have change my code accordingly. @jenny-alexander please approve my PR, or let me know if I need to do extra work.

@MarynaRomashca - can you please check the "read" status when adding a new book? It seems to be doing the opposite of what is intended.
- update the changelist in the Pull Request to briefly explain what this PR does.

MarynaRomashca commented:
@jenny-alexander Thanks for your time. Done, I have up dated the "read" status in the right direction.

jenny-alexander reviewed (approved):
Approving but pointing out that the changelist portion is still not updated.

MarynaRomashca commented:
@jenny-alexander Thanks for aprroval. Sorry, I don't understend, where is a ChangeList? How to find it? 
I can just write you there: 
My change list: 
1. Converted unnecessary "let" declarations to "const" for better readability and immutability.
2. Moved DOM element selections (title, author, pages, check) inside the submit() function to limit their scope and reduce global variables.
3. Simplified code form validation using if (!title.value || !pages.value).
4. Improved render() function to clear and rebuild table correctly.
5. Updated read/unread toggle button logic with instant re-render.

cjyuan commented:
@MarynaRomashca This "Changelist" 

![image](https://github.com/user-attachments/assets/742c700f-df8d-4750-99e0-7a8a14ab63c9)
